keanu-project/src/main/java/io/improbable/keanu/tensor/TensorShape.java

```diff
@@ -248,4 +259,31 @@ public static int getAbsoluteDimension(int dimension, int rank) {
         return dimension;
     }
 
+    public static long[] getSummationResultShape(long[] inputShape, int[] sumOverDimensions) {
+        if (inputShape.length > 0) {
+            return ArrayUtils.removeAll(inputShape, sumOverDimensions);
+        } else {
+            Preconditions.checkArgument(sumOverDimensions.length == 0);
+            return inputShape;
+        }
+    }
+
+    public static long[] getPermutedResultShapeShape(long[] shape, int... rearrange) {
+        long[] permutedShape = new long[shape.length];
+        for (int i = 0; i < shape.length; i++) {
+            permutedShape[i] = shape[rearrange[i]];
+        }
+        return permutedShape;
+    }
+
+    public static int[] invertedPermute(int[] rearrange) {
+        int[] inverted = new int[rearrange.length];
+
+        for (int i = 0; i < rearrange.length; i++) {
+            inverted[rearrange[i]] = i;
+        }
+
+        return inverted;
+    }
```

Comment:
Agree it would be better to explicitly cover them with UT, but checking coverage from our existing tests, it's only exceptions (5 lines) that aren't covered currently by existing tests.  Given time constraints, I'd suggest we raise a low prio task for this?